src/Foundatio.Parsers.LuceneQueries/Nodes/GroupNode.cs

```diff
@@ -91,7 +91,7 @@ public class GroupNode : QueryNodeBase, IFieldQueryWithProximityAndBoostNode {
             builder.Append("(");
 
         if (Left != null)
-            builder.Append(Left);
+            builder.Append(Left is GroupNode ? ((GroupNode) Left).ToString(defaultOperator) : Left.ToString());
```

Comment:
It's a bit ugly using a cast here and below but it's the simplest option.
An alternative is to move this entire method back into the GenerateQueryVisitor

Reply:
Yeah, I think it's fine.